src/api/server/sync_io.rs

```diff
@@ -121,7 +121,11 @@ impl<F: FileSystem + Sync> Server<F> {
 
     fn lookup<S: BitmapSlice>(&self, mut ctx: SrvContext<'_, F, S>) -> Result<usize> {
         let buf = ServerUtil::get_message_body(&mut ctx.r, &ctx.in_header, 0)?;
-        let name = bytes_to_cstr(buf.as_ref())?;
+        let name = bytes_to_cstr(buf.as_ref()).map_err(|e| {
+            let _ = ctx.reply_error_explicit(io::Error::from_raw_os_error(libc::EINVAL));
```

Comment:
I think it's better to print out the original error, so we know where the error happens exactly and why.

Reply:
ok